### ISSUE TITLE
Can the generated UML function with arguments ?

### ISSUE THREAD

**Issue by 625781186** [doxygen/doxygen#7686]
![image](https://user-images.githubusercontent.com/30521429/78316150-862c9d80-7591-11ea-93cb-f9f745cbf303.png)

**Comment by albert-github** [comment]
I think the problem here will be that the diagrams will become very fast unreadable, due to the length of the argument list.
I made a very very quick test with a png and svg image (I cannot add the svg image here so it will be in an attachment):

![class_q_string__inherit__graph](https://user-images.githubusercontent.com/5223533/78355877-d1a17480-75ae-11ea-80b4-f237e4c97b10.png)

and the original image:

![class_q_string__inherit__graph](https://user-images.githubusercontent.com/5223533/78355901-d9f9af80-75ae-11ea-9ebd-bf86770d923b.png)

We see in the images that some "strange" ways the lines are folded (especially in the qdict images).
With the svg images we see even that there is an overflow of the boxes (also a bit depending on the zoom level). As it was a very quick test there might be some possibilities to improve the box sizes etc.

The change was just changing (in dotnode line 151 in function writeBoxMemberList) from:
```
          t << prot << " ";
          t << DotNode::convertLabel(mma->name());
          if (!mma->isObjCMethod() &&
            (mma->isFunction() || mma->isSlot() || mma->isSignal())) t << "()"; 
          t << "\\l";
```
into
```
          t << prot << " ";
          t << DotNode::convertLabel(mma->typeString());
          t << " ";
          t << DotNode::convertLabel(mma->name());
          if (!mma->isObjCMethod() &&
            (mma->isFunction() || mma->isSlot() || mma->isSignal())) t << DotNode::convertLabel(mma->argsString());
          t << "\\l";
```

When implementing the arguments some attention should be paid to:
- it should probably behind an `UML_WITH_ARGS` configuration setting.
- line breaking
- box overflow

@doxygen what do you think about it.

All images: [images.tar.gz](https://github.com/doxygen/doxygen/files/4426936/images.tar.gz)

**Comment by 625781186** [comment]
@albert-github
Hi , 
I can live with some typographical flaws, 
so how do I generate function parameters?

**Comment by albert-github** [comment]
> I can live with some typographical flaws,

In the doxygen this would not be good.

> so how do I generate function parameters?

What I did was just some quick and dirty hack in the code as indicated. But I'm waiting for some more input about the pitfalls I might have overlooked.

**Comment by 625781186** [comment]
> > I can live with some typographical flaws,
> 
> In the doxygen this would not be good.
> 
> > so how do I generate function parameters?
> 
> What I did was just some quick and dirty hack in the code as indicated. But I'm waiting for some more input about the pitfalls I might have overlooked.

Will this feature be added in the future , I need it.

**Comment by 625781186** [comment]
@albert-github 
Hi ,  Is there any progress on this issue ? 
I need It.

**Comment by albert-github** [comment]
At the moment there is no progress, waiting on "@doxygen what do you think about it.".
Personally I'm not really in favor of it.

You can of course patch your own version in the way I suggested (no guarantee that it will work in future or that it has no side effects).

**Comment by 625781186** [comment]
This code in  picture  is the way you suggested ,did I understand that correctly?

![image](https://user-images.githubusercontent.com/30521429/80596582-ce37c680-8a50-11ea-98d5-b35eeaa60a18.png)

**Comment by albert-github** [comment]
Indeed with the limitations I mentioned in my previous comment.

**Comment by doxygen** [comment]
> At the moment there is no progress, waiting on "@doxygen what do you think about it.".
> Personally I'm not really in favor of it.

Me neither. The picture would just become too big to be useful, and you would need to click it anyway to see the details.
 
Looking at the picture above without arguments, I so see some room for improvement: collapse all overloaded methods into one, and use `...` if the method has arguments, and maybe add a counter behind it, i.e. `QString(...) [7]`, instead of `7x QString()`...

**Comment by 625781186** [comment]
> > At the moment there is no progress, waiting on "@doxygen what do you think about it.".
> > Personally I'm not really in favor of it.
> 
> Me neither. The picture would just become too big to be useful, and you would need to click it anyway to see the details.
> 
> Looking at the picture above without arguments, I so see some room for improvement: collapse all overloaded methods into one, and use `...` if the method has arguments, and maybe add a counter behind it, i.e. `QString(...) [7]`, instead of `7x QString()`...

No , I don't think so.
So far , I use doxygen  generated UML to 'svg' and input it to visio.exe , after  I will link uml . 
So , the picture is too big I can accept, but I can't accept without parameters.

I think better mathod  as https://github.com/doxygen/doxygen/issues/7686#issuecomment-608384636 . 
Add an  `UML_WITH_ARGS` configuration setting for the user to choose.
![image](https://user-images.githubusercontent.com/30521429/80685632-10691280-8afa-11ea-96a3-2518caa782fe.png)

**Comment by albert-github** [comment]
The mentioned pull request has been integrated into master and released with version 1.9.0, so the problem has been solved in tyhe mean time (this issue will be automatically closed with the next release).
